storage/localstore/mode_set_test.go

```diff
@@ -63,180 +64,341 @@ func TestModeSetAccess(t *testing.T) {
-			pin:             false,
-			expErrPushIndex: leveldb.ErrNotFound, // not in push index
-			expErrGCIndex:   nil,                 // GCd
-			expErrPinIndex:  leveldb.ErrNotFound, // not pinned
-		},
-		// tag IS anonymous, set pull synced WILL cause it to go
-		// into gc index, it SHOULD be in the pin index
-		{
-			name:            "set pull sync, anonymous tag, with pinning",
-			mode:            chunk.ModeSetSyncPull,
-			anonymous:       true,
-			pin:             true,
-			expErrPushIndex: leveldb.ErrNotFound, // not in push index
-			expErrGCIndex:   nil,                 // GCd
-			expErrPinIndex:  nil,                 // is pinned
-		},
-		// tag IS NOT anonymous, set push synced WILL cause it to go
-		// into gc index, it SHOULD NOT be in the pin index
-		{
-			name:            "set push sync, normal tag, no pinning",
-			anonymous:       false,
-			pin:             false,
-			mode:            chunk.ModeSetSyncPush,
-			expErrPushIndex: leveldb.ErrNotFound, // not in push index
-			expErrGCIndex:   nil,                 // GCd
-			expErrPinIndex:  leveldb.ErrNotFound, // not pinned
-		},
-		// tag IS NOT anonymous, set push synced WILL cause it to go
-		// into gc index, it SHOULD be in the pin index
-		{
-			name:            "set push sync, normal tag, with pinning",
-			anonymous:       false,
-			pin:             true,
-			mode:            chunk.ModeSetSyncPush,
-			expErrPushIndex: leveldb.ErrNotFound, // not in push index
-			expErrGCIndex:   nil,                 // GCd
-			expErrPinIndex:  nil,                 // is pinned
-		},
-		// tag IS anonymous, set push synced WILL cause it to go
-		// into gc index, it SHOULD NOT be in the pin index
-		{
-			name:            "set push sync, anonymous tag, no pinning",
-			anonymous:       true,
-			pin:             false,
-			mode:            chunk.ModeSetSyncPush,
-			expErrPushIndex: nil,                 // not in push index
-			expErrGCIndex:   leveldb.ErrNotFound, // not GCd
-			expErrPinIndex:  leveldb.ErrNotFound, // not pinned
-		},
-		// tag IS anonymous, set push synced WILL NOT cause it to go
-		// into gc index, it SHOULD be in the pin index
-		{
-			name:            "set push sync, anonymous tag, with pinning",
-			anonymous:       true,
-			pin:             true,
-			mode:            chunk.ModeSetSyncPush,
-			expErrPushIndex: nil,                 // is in push index
-			expErrGCIndex:   leveldb.ErrNotFound, // not GCd
-			expErrPinIndex:  nil,                 // is pinned
-		},
-	} {
-		t.Run(mtc.name, func(t *testing.T) {
-			for _, tc := range multiChunkTestCases {
-				t.Run(tc.name, func(t *testing.T) {
-					db, cleanupFunc := newTestDB(t, &Options{Tags: chunk.NewTags()})
-					defer cleanupFunc()
-
-					tag, err := db.tags.Create(mtc.name, int64(tc.count), mtc.anonymous)
-					if err != nil {
-						t.Fatal(err)
-					}
-					if tag.Uid != 0 {
-						t.Fatal("expected mock tag uid")
-					}
+// here we try to set a normal tag (that should be handled by pushsync)
```

Comment:
It looks to me that there some code deduplication can be done in new tests. The previous test had too many variables for subtests, but maybe some test helpers can reduce the duplication in tests. Just an idea.

Reply:
removed some of the test code with the `chunk/testing#CheckTag` helper function. not sure if there's anything else to remove here. i find this manageable to read